spec/factories/service_template.rb

```diff
@@ -4,4 +4,18 @@
   factory :service_template_ansible_playbook, :class => 'ServiceTemplateAnsiblePlaybook', :parent => :service_template
   factory :service_template_container_template, :class => 'ServiceTemplateContainerTemplate', :parent => :service_template
   factory :service_template_transformation_plan, :class => 'ServiceTemplateTransformationPlan', :parent => :service_template
+
+  trait :with_provision_resource_action_and_dialog do
+    after(:create) do |x|
+      x.resource_actions << FactoryGirl.create(:resource_action, :with_dialog, :action => 'Provision')
+    end
+  end
+
+  trait :orderable do
+    display true
+
+    after(:create) do |x|
+      FactoryGirl.create(:service_template_catalog, :service_templates => [x])
```

Comment:
You may or may not need/want to assign this to the initialized object if things are wonky, check to make sure `created_service_template.service_template_catalog` works. If it doesn't you need to assign it here.

Reply:
@chrisarcand it does ✨ I had it the other way (which didn't work)